File: backend/src/main/java/com/kosa/recruit/domain/entity/Weight.java
package com.kosa.recruit.domain.entity;

import lombok.*;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "weight")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Weight {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "target_type", nullable = false)
    private String targetType;
    
    @ManyToOne
    @JoinColumn(name = "jobpost_id",
        foreignKey = @ForeignKey(name = "fk_weight_jobpost"))
    private JobPost jobPost;
    
    @Column(name = "field_name", nullable = false)
    private String fieldName;
    
    @Column(name = "weight_value")
    private Float weightValue;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: backend/app/models/recruitment/weight.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.sql import func
from app.core.database import Base

class Weight(Base):
    __tablename__ = "weight"
    
    id = Column(Integer, primary_key=True, index=True)
    target_type = Column(String(255), nullable=False)
    jobpost_id = Column(Integer, ForeignKey("jobpost.id"), nullable=True)
    field_name = Column(String(255), nullable=False)
    weight_value = Column(Float, nullable=True)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now()) 